src/tracing/mod.rs

```diff
@@ -391,7 +398,7 @@ impl TracingInspector {
 
         // The gas cost is the difference between the recorded gas remaining at the start of the
         // step the remaining gas here, at the end of the step.
-        step.gas_cost = step.gas_remaining - self.gas_inspector.gas_remaining();
+        // step.gas_cost = step.gas_remaining - self.gas_inspector.gas_remaining();
```

Comment:
hard to find what's happening inside revm here

but this here would also overflow if nut saturating:

https://github.com/bluealloy/revm/blob/d6bd0109e5b5c37505fe25017b8ca5c2ea5ae6ca/crates/revm/src/inspector/gas.rs#L44

unclear why, I assume it must have something to do with a RETURN,
because this overflow happens after a call ended (`call_end`)

any ideas @rakita 

@DaniPopes I think for now we can just saturate, even if not accurate

Reply:
call_end returned the gas that is not spend to the caller after sub call get returned.

Reply:
But with loop calls this behaviour is changed. It should not be effected. So overflow can only happen is step_end has more `remaining_gas` than `step` and step is always first called.